src/features/swap/SwapForm.tsx

```diff
@@ -124,17 +129,24 @@ function SwapFormInputs({ balances }: { balances: AccountBalances }) {
     }
   }, [setFieldValue, chain, values, swappableTokenOptions, isConnected])
 
-  const roundedBalance = fromWeiRounded(balances[fromTokenId], Tokens[fromTokenId].decimals)
+  const roundedBalance = fromWeiRounded(
+    balances[fromTokenId as keyof AccountBalances],
+    Tokens[fromTokenId as TokenId].decimals
+  )
   const isRoundedBalanceGreaterThanZero = Boolean(Number.parseInt(roundedBalance) > 0)
-  const onClickUseMax = () => {
-    const maxAmount = fromWei(balances[fromTokenId], Tokens[fromTokenId].decimals)
-    setFieldValue('amount', maxAmount)
-
-    setFieldValue('direction', 'in')
-    if (fromTokenId === TokenId.CELO) {
-      toast.warn('Consider keeping some CELO for transaction fees')
+  const onClickUseMax = useCallback(() => {
+    const maxAmount = fromWei(
+      balances[fromTokenId as keyof AccountBalances],
+      Tokens[fromTokenId as TokenId].decimals
+    )
+    if (values.amount !== maxAmount) {
+      setFieldValue('amount', maxAmount)
+      setFieldValue('direction', 'in')
+      if (fromTokenId === TokenId.CELO) {
```

Comment:
Would we show something in case of some token besides CELO? We can show some warn message about taking the max token amount. It'd be like this:
`
    if (values.amount !== maxAmount) {
      setFieldValue('amount', maxAmount)
      setFieldValue('direction', 'in')
      toast.warn(`${fromTokenId === TokenId.CELO ? "Consider keeping some CELO for transaction fees" : "Taking all for swapping"}`)
`

Reply:
No additional requirements to apply - can be considered as an improvement